### ISSUE TITLE
Assumes returned IQueryable directly holds dbContext's entities

### ISSUE THREAD

**Issue by jpa00** [trenoncourt/AutoQueryable#1]
Currently, the code relies on the IQueryable having an entity type, so the the properties can be listed from the dbContext. This prevents using mapped objects, for example by utilizing AutoMapper's ProjectTo, which would otherwise be able to map any expressions AutoQueryable would produce to the underlying entities. Something like:
`return myService.Get().ProjectTo<ExampleDTO>()`

So I was wondering if there's a reason you're digging the properties from dbContext and not using reflection? (Although reflection might make the code a bit slower, maybe.) I'm asking since Asp.Net Core OData support is not usable at this point and AutoQueryable has already nearly everything that I'd need... And I would prefer to not create a very similar piece of code with the small difference of using reflection instead of relying on the dbContext. :)

**Comment by trenoncourt** [comment]
Hi jpa00,
Using Dto projection is on the roadmap and I've started to work on it yesterday.

Dto projection will be supported by two ways :
- Passing the dto type to the autoqueryable attribute and Autoqueryable do the work for mapping.
- Return projection like your example (the simplest way for me)

I will try to push the second by the end of the week.
For now you can une projection with the select keyword :
http://baseurl/api/products?select=id,color

That create type with IL and use reflection to map to the new type. :).

**Comment by trenoncourt** [comment]
Hi jpa00,
I just pushed a new version (0.9.0) with Dto projection capability.

```c#
[HttpGet]
[AutoQueryable]
public IQueryable Get([FromServices] AdventureWorksContext adventureWorksContext)
{
    return adventureWorksContext.Product.Select(p => new ProductProjection
    {
        Name = p.Name
    });
}

```

Note that with Dto projection, you can't use include keywords.

I have not made a lot of tests but it seems to work ;).

**Comment by jpa00** [comment]
Good point about not being able to use includes after the projection, although in my (our) case that's most likely a bonus. I'll check out the new version in a couple hours.

**Comment by jpa00** [comment]
Filtering and ordering work beautifully, but take and skip crash & burn. :) I get an exception saying something like:
`"'Microsoft.EntityFrameworkCore.Query.Internal.EntityQueryable<myDTO>' does not contain a definition for 'Take'"`

I don't have time yet to check deeper into what causes this. I know I can use both Take and Skip on the Queryable directly though.

**Comment by jpa00** [comment]
Hey, I know this isn't exactly the right place to ask about this, but since we're talking about take and skip... Have you thought about maybe optionally wrapping the result set in an object to add a count and possibly a next-link odata-style?

**Comment by trenoncourt** [comment]
Hi jpa00,
I fixed Take, Skip, First, Last in version 0.9.1, I have not a lot of time to test, tell me if you saw an error.

I will create a new thead about wrapping the object
